app/src/main/java/com/arctouch/codechallenge/data/MovieDataSource.java

```diff
@@ -0,0 +1,99 @@
+package com.arctouch.codechallenge.data;
+
+import android.arch.paging.PageKeyedDataSource;
+import android.support.annotation.NonNull;
+
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.network.TmdbApi;
+import com.arctouch.codechallenge.network.RetrofitClient;
+
+import java.util.ArrayList;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class MovieDataSource extends PageKeyedDataSource<Integer, Movie> {
```

Comment:
Nice that you used Paging Library! 👏

Reply:
Thanks for reviewing my code, any comments and improvements are welcome.